src/terminal/Line.h

```diff
@@ -235,6 +235,19 @@ class Line
         return inflatedBuffer().at(unbox<size_t>(column)).width();
     }
 
+    [[nodiscard]] bool compareCellTextAt(ColumnOffset column, char codepoint) const noexcept
```

Comment:
Please already assume that `TrivialLineBuffer` already contains UTF-8.
Currently it's only containing US-ASCII, but with a future PR, that will be backwards-compatibly UTF-8.

Assuming that this function is **only** being invoked when snapping interactive selections, I would say it is okay to inflate the underlying buffer in order to easily access the grapheme cluster at a given column. That makes it more expensive for the first call, but faster for further calls, as users usually don't stop the mouse-move as soon as it snaps. - The general case of improved performance for trivial lines is also still retained :)

With that, I think this input parameter should at least be `char32_t` (or `u32string_view`).